merkle/hashers/tree_hasher.go

```diff
@@ -42,3 +48,47 @@ type MapHasher interface {
 	// TODO(gbelvin): Replace Size() with BitLength().
 	Size() int
 }
+
+var (
+	maxHashers = trillian.HashStrategy(len(trillian.HashStrategy_value))
+	logHashers = make([]LogHasher, maxHashers)
+	mapHashers = make([]MapHasher, maxHashers)
+)
+
+// RegisterLogHasher registers a hasher for use.
+func RegisterLogHasher(h trillian.HashStrategy, f LogHasher) {
+	if h < 0 || h >= maxHashers {
+		panic(fmt.Sprintf("RegisterLogHasher(%v) of unknown hasher", h))
```

Comment:
- We're checking for `UNKNOWN_HASH`.  
- `hasher == nil` is ok, and will be caught when requested. 
- I'll add a check for overrides.